### PR TITLE
Add product block resource to graphql root Query

### PR DESCRIPTION
**Pull request by tjeerddie**
Closes: #303

### CODE REVIEW

File: orchestrator/graphql/resolvers/product_block.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+async def resolve_product_blocks(
+    info: CustomInfo,
+    filter_by: Union[list[GraphqlFilter], None] = None,
+    sort_by: Union[list[GraphqlSort], None] = None,
+    first: int = 10,
+    after: int = 0,
+) -> Connection[ProductBlock]:
+    _error_handler = create_resolver_error_handler(info)
+
+    pydantic_filter_by: list[Filter] = [item.to_pydantic() for item in filter_by] if filter_by else []
```
Comment (mrijk) [inline orchestrator/graphql/resolvers/product_block.py:28]: Just curious, do why do we prefix these variables with pydantic? Is that relevant?

**Review by mrijk — APPROVED** [review]
LGTM

File: orchestrator/graphql/resolvers/product_block.py (hunk 1 above)
Comment (tjeerddie) [reply orchestrator/graphql/resolvers/product_block.py:28]: to not overwrite `filter_by`.

File: orchestrator/db/sorting/product_block.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from sqlalchemy.inspection import inspect
+
+from orchestrator.db import ProductBlockTable
+from orchestrator.db.sorting.sorting import generic_column_sort, generic_sort
+from orchestrator.utils.helpers import to_camel
+
+PRODUCT_BLOCK_SORT_FUNCTIONS_BY_COLUMN = {
+    to_camel(key): generic_column_sort(value) for [key, value] in inspect(ProductBlockTable).columns.items()
```
Comment (Mark90) [inline orchestrator/db/sorting/product_block.py:8]: List syntax is not needed here

```suggestion
    to_camel(key): generic_column_sort(value) for key, value in inspect(ProductBlockTable).columns.items()
```

File: orchestrator/graphql/resolvers/product_block.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    pydantic_sort_by: list[Sort] = [item.to_pydantic() for item in sort_by] if sort_by else []
+    logger.info(
```
Comment (Mark90) [inline orchestrator/graphql/resolvers/product_block.py:30]: decrease log level to debug?

File: orchestrator/graphql/resolvers/product_block.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    return Connection(
+        page=page_product_blocks,
+        page_info=PageInfo(
+            has_previous_page=bool(after),
+            has_next_page=has_next_page,
+            start_cursor=start_cursor,
+            end_cursor=end_cursor,
+            total_items=total if total else None,
+        ),
```
Comment (Mark90) [inline orchestrator/graphql/resolvers/product_block.py:56]: I think we'll see this pattern (requesting an extra item to determine has_next_page, calculating the cursors etc) in a few more places (if not already).

We could probably make a generic function for line 36-56? Fine with me to do this in the next PR though!

File: orchestrator/graphql/resolvers/product_block.py (hunk 3 above)
Comment (tjeerddie) [reply orchestrator/graphql/resolvers/product_block.py:30]: The logging was kept from the old filtering, I don't mind changing it

**Review by Mark90 — APPROVED** [review]

File: orchestrator/graphql/resolvers/product_block.py (hunk 3 above)
Comment (Mark90) [reply orchestrator/graphql/resolvers/product_block.py:30]: With info it's going to be really noisy in production :)

**Pull request merged**